fields/field.reflection.php

```diff
@@ -385,9 +385,38 @@ public function compile(&$entry)
             // Find queries:
             preg_match_all('/\{[^\}]+\}/', $expression, $matches);
 
-            // Find replacements:
+            // Get root node name
+            $root_node_name = preg_quote ((string)$xpath->evaluate('name(/*)'), '#');
+
+            // Find replacements
+            // including modifications for edge-cases (#35)
             foreach ($matches[0] as $match) {
-                $result = @$xpath->evaluate('string(' . trim($match, '{}') . ')');
+
+                // Add a leading '/' if the expression starts with the root node name
+                // Fixes expressions 1.J and 2.B
+                if (preg_match('#^{'.$root_node_name.'#', $match)) {
+                    $result = @$xpath->evaluate('string(/' . trim($match, '{}') . ')');
+                }
+
+                // Insert a '/' if the expression contains a '(' directly followed by the root node name
+                // Fixes expression 1.J (when used inside an xpath-function)
+                else if (strpos($match,'('.$root_node_name) !== false) {
```

Comment:
Sorry about that, we need to keep $root_node_name intact to make the strpos work... !!

```php
$root_node_name = (string)$xpath->evaluate('name(/*)')
$root_node_name_quoted = preg_quote ($root_node_name, '#'); // <-- use in preg_* calls only
```

Reply:
Implemented it as suggested – but as I'm really not an expert regarding all this preg/regex-stuff, I would be glad if you could give me a hint what kind of difference exists between my previous solution…

    preg_match('#^{'.$root_node_name.'#', $match))

and the one we have now:

    preg_match('#^{'.$root_node_name_quoted.'#', $match))

I mean both work as expected in all situations I tested, but I'd really like to understand that change :)